### ISSUE TITLE
Broken report barcode for records with special characters

### ISSUE THREAD

**Issue by themreza** [odoo/odoo#76590]
Impacted versions: 14.0

Steps to reproduce:

1. Edit the warehouse short name to include an accented  `ÖDOO` (this is applicable to other records and reports as well)
2. Create a sale order with a delivery move
3. Print the Picking Operations report

Current behavior:

The barcode image is broken
![barcode-broken](https://user-images.githubusercontent.com/5465057/133447727-c7142287-4df2-4ca6-a42b-61ba6b18befd.png)

Expected behavior:

The barcode should be displayed correctly.

Video/Screenshot link (optional):

https://user-images.githubusercontent.com/5465057/133447778-fe2f5216-902b-4681-8d1c-592d4ff34986.mp4

Technical hint:

This works: https://9970963-14-0-all.runbot69.odoo.com/report/barcode/?type=Code128&value=ODOO/PICK/11284&width=600&height=100

This doesn't: https://9970963-14-0-all.runbot69.odoo.com/report/barcode/?type=Code128&value=%C3%96DOO/PICK/11284&width=600&height=100

The bug occurs here: https://github.com/odoo/odoo/blob/14.0/addons/stock/report/report_stockpicking_operations.xml#L12

I see that some other barcodes already have some solution to escape special characters: https://github.com/odoo/odoo/blob/14.0/addons/stock_picking_batch/report/report_picking_batch.xml#L17

**Comment by lem8r** [comment]
Barcode of type Code128 supports only ASCII characters.

**Comment by themreza** [comment]
Indeed. One solution is to change this barcode optionally to a QR code. But either way, it would be helpful to warn users when changing fields that can be used to generate barcodes.

**Comment by RicardCForgeFlow** [comment]
Hello! This issue still happens in master and 19.0
Maybe one option should be to raise a ValidationError like:
<img width="1917" height="891" alt="Image" src="https://github.com/user-attachments/assets/7c785016-619c-4e08-95f7-8ae1cf6adbc9" />
